system/jlib/jutil.cpp

```diff
@@ -2868,29 +2868,103 @@ int parseCommandLine(const char * cmdline, MemoryBuffer &mb, const char** &argvo
     return 0;
 }
 
+#ifndef _WIN32
+# define ALL_PERMS (S_IRUSR|S_IWUSR|S_IXUSR|S_IRGRP|S_IWGRP|S_IXGRP|S_IROTH|S_IWOTH|S_IXOTH)
+#endif
+
 jlib_decl StringBuffer &getTempFilePath(StringBuffer & target, const char * component, IPropertyTree * pTree)
 {
     StringBuffer dir;
     if (pTree)
         getConfigurationDirectory(pTree->queryPropTree("Directories"),"temp",component,pTree->queryProp("@name"),dir);
-    if (!dir.length())
+    bool ok = false;
+    if (!dir.isEmpty())
     {
+        ok = recursiveCreateDirectory(dir.str());
+        if (ok)
+        {
 #ifdef _WIN32
-        char path[_MAX_PATH+1];
-        if(GetTempPath(sizeof(path),path))
-            dir.append(path).append("HPCCSystems\\hpcc-data");
-        else
-            dir.append("c:\\HPCCSystems\\hpcc-data\\temp");
+            return target.set(dir);
 #else
-        dir.append(getenv("TMPDIR"));
-        if (!dir.length())
-            dir.append(RUNTIME_DIR);
-        dir.append("/hpcc-data/temp");
+            int srtn = access(dir.str(), R_OK|W_OK|X_OK);
+            if (!srtn)
+            {
+                chmod(dir.str(), ALL_PERMS);
```

Comment:
Should we open up perms on these temp dirs ?

Reply:
Agree, but it goes to the source of this issue.  If, when running jobs as the hpcc user, the temp dir was opened up, then running as a different user would have worked ok.  Without this we fall through to the final logic block which creates a unique temp dir for every user.

Reply:
@richardkchapman

Reply:
Allowing other users access to the hpcc temp directory is definitely NOT what we want. We want users to use temp directories that they have access to, not to give them access to whatever directory they happen to feel like using.

You could argue that we should not try to use the hpcc temp directory at all if the user is not hpcc (where by "hpcc" I mean the value of RUNTIME_USER macro), even if by some misconfiguration we did have access to it.

Reply:
@mckellyln

Reply:
Updated with second commit to remove chmod() calls.